services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -255,8 +255,13 @@ const deleteIndexedBlocks = async job => {
 						transactions,
 						async (tx) => {
 							// Invoke 'revertTransaction' to execute command specific reverting logic
-							await revertTransaction(blockHeader, tx, events, dbTrx);
 							const normalizedTransaction = await normalizeTransaction(tx);
+
```

Comment:
On L247, we can directly fetch all the transactions by blockID using the `getTransactionsByBlockID` method defined in `services/blockchain-indexer/shared/dataService/business/transactions.js`.

Also on L249, we need to fetch the events from the cache determined by the blockID instead of the blockHeight since the node will have replaced the events corresponding the new block at the same height.

Reply:
Applied on L247.
Currently we dont support getting events by blockID so it will be handled in separate issue.